src/sentry/static/sentry/app/components/stream/group.jsx

```diff
@@ -135,9 +135,13 @@ const StreamGroup = createReactClass({
 
     if (filtered && query) {
       const queryObj = queryToObj(query);
-      for (const key in queryObj)
-        if (!['is', '__text'].includes(key))
-          discoveryQueryTerms.push(`${key}:${queryObj[key]}`);
+      for (const queryTag in queryObj)
+        if (!['is', '__text'].includes(queryTag)) {
+          const queryVal = queryObj[queryTag].includes(' ')
+            ? `"${queryObj[queryTag]}"`
+            : queryObj[queryTag];
+          discoveryQueryTerms.push(`${queryTag}:${queryVal}`);
+        }
```

Comment:
Do you know if this is logic that's used elsewhere in the app? cc @markstory

Reply:
There are other locations that link to discovery with modified query but this is a unique use case. Maybe I can make a `utils/stream` function that picks out tags from query objects, but that felt like an overkill

Reply:
We have a few places that manipulate queries. Most of those use the tokenizeSearch module. The QueryResult object in that module lets you add and remove tags and gives you a better string conversion API that handles quoting better. You'd still need to drop the `is` condition though.